LiteCore/tests/c4BaseTest.cc

```diff
@@ -139,6 +141,33 @@ TEST_CASE("C4Error Reporting Macros", "[Errors][C]") {
 #endif
 }
 
+TEST_CASE_METHOD(C4Test, "Create collection concurrently", "[Database][C]") {
+    const slice             dbName = db->getName();
+    const C4DatabaseConfig2 config = db->getConfiguration();
+
+    c4::ref db2 = c4db_openNamed(dbName, &config, ERROR_INFO());
+    REQUIRE(db2);
+
+    char buf[6]{};
+    for ( int i = 0; i < 5; i++ ) {
+        C4Error err{};
+        C4Error err2{};
+
+        snprintf(buf, 6, "coll%i", i);
+
+        {
+            slice                  collName{buf};
+            const C4CollectionSpec spec{collName, "scope"_sl};
+
+            auto a1 = std::async(std::launch::async, c4db_createCollection, db, spec, ERROR_INFO(&err));
```

Comment:
Access the same db concurrently is not legit anyway

Reply:
Jianmin is right — LiteCore's API is intentionally not thread-safe. You can’t use a C4Database concurrently on two threads.

I don’t think there’s a good test for this, honestly. You’d have to repeat the test many (hundreds?) times in a tight loop, creating a new database each time, and even then you might not hit the race condition.